resources/sdk/purecloudjavascript/templates/README.mustache

```diff
@@ -160,6 +160,71 @@ client.refreshCodeAuthorizationGrant(clientId,clientSecret,refreshToken)
   });
 ```
 
+**Node.js** [PKCE Grant](https://developer.genesys.cloud/authorization/platform-auth/use-pkce)
+
+```javascript
+const client = {{moduleName}}.ApiClient.instance;
+client.authorizePKCEGrant(clientId,codeVerifier,authCode,redirectUri)
+  .then(() => {
+    // Do authenticated things
+  })
+  .catch((err) => {
+    // Handle failure response
+    console.log(err);
+  });
+```
+
+The SDK provides methods to generate a PKCE Code Verifier and to compute PKCE Code Challenge.
+
+```javascript
+let codeVerifier = client.generatePKCECodeVerifier(128);
+let codeChallenge = client.computePKCECodeChallenge(codeVerifier);
+```
+
+**Web** [PKCE Grant](https://developer.genesys.cloud/authorization/platform-auth/use-pkce)
+
+The _loginPKCEGrant_ only works when used in a browser. This is because a node.js application does not have a browser interface that can display the Genesys Cloud login window.
+
+Optional parameters may be specified in the optional third parameter for `loginPKCEGrant`. This parameter accepts an object with key/value pairs. Supported properties:
+
+* `state` - An arbitrary string used to associate a login request with a login response. This value will be provided in the `state` property on the object when the promise is resolved. The state in the resolved promise will be identical to what was passed into `loginPKCEGrant`, except when the state is retrieved from the auth query upon completing a login; in that case, the state from the auth query will override the passed in state.
+
+The _loginPKCEGrant_ supports an optional fourth parameter for `loginPKCEGrant`. This parameter accepts a string, used to provide a code verifier as input. When no code verifier is provider (Method 1), the SDK automatically generates a PKCE Code Verifier and saves it in _window sessionStorage_. If a code verifier is provided (Method 2), it is up to the custom application to store the code verifier value and pass it in _loginPKCEGrant_.
+
+```javascript
+const client = {{moduleName}}.ApiClient.instance;
+
+// Method1: Let loginPKCEGrant generate the code verifier
+client.loginPKCEGrant(clientId, redirectUri, { state: state })
+  .then((data) => {
+    console.log(data);
+    // Do authenticated things
+  })
+  .catch((err) => {
+    // Handle failure response
+    console.log(err);
+  });
+
+// Method2: code verifier as input parameter
+let codeVerifier = client.generatePKCECodeVerifier(128);
+client.loginPKCEGrant(clientId, redirectUri, { state: state }, codeVerifier)
+  .then((data) => {
+    console.log(data);
+    // Do authenticated things
+  })
+  .catch((err) => {
+    // Handle failure response
+    console.log(err);
+  });
+```
+
+The SDK provides methods to generate a PKCE Code Verifier and to compute PKCE Code Challenge.
+
+```javascript
```

Comment:
same for here, probably best to put this in the code block above to keep everything together